File: frontend/src/types/report.ts
import type { BehaviorCategory } from './trace'

export type Verdict = 'SAFE' | 'SUSPICIOUS' | 'DANGEROUS'

export interface BehaviorEntry {
  category: BehaviorCategory
  name: string
  description: string
  risk_score: number
  mitre_id: string
  mitre_name: string
  syscalls: string[]
}

export interface Concept {
  term: string
  definition: string
  course_topic: string
}

export interface AIReport {
  verdict: Verdict
  confidence: number
  narrative: string
  behaviors: BehaviorEntry[]
  concepts: Concept[]
}

File: backend/app/schemas/report.py
from __future__ import annotations
from typing import Literal
from pydantic import BaseModel

from app.schemas.trace import BehaviorCategory

Verdict = Literal["SAFE", "SUSPICIOUS", "DANGEROUS"]


class BehaviorEntry(BaseModel):
    category: BehaviorCategory
    name: str
    description: str
    risk_score: int
    mitre_id: str
    mitre_name: str
    syscalls: list[str]


class Concept(BaseModel):
    term: str
    definition: str
    course_topic: str


class AIReport(BaseModel):
    verdict: Verdict
    confidence: int
    narrative: str
    behaviors: list[BehaviorEntry]
    concepts: list[Concept]
